torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -302,10 +302,8 @@ static void cacheAllocatorRegisterHook(
   }
 
   std::lock_guard<std::mutex> lock(ncclCommDevIdxMapMutex);
-  for (auto& it : ncclCommDevIdxMap) {
-    auto& ncclComm = it.first;
-    auto& devIdx = it.second;
-    if (te.device_ == devIdx) {
+  for (auto& [ncclComm, _] : ncclCommDevIdxMap) {
+    if (te.device_ == ncclComm->getDeviceIndex()) {
```

Comment:
Note that with this PR we query the device from the `NCCLComm` (the key of the hashmap) instead using the value of the hashmap. After this PR, the values of the hashmap are _never_ used _anywhere_. This is on purpose, since in a later PR I plan to repurpose that hashmap in order to store something different in it.

Reply:
What do you use the `ncclCommDevIdxMap` in your later PR? (Sorry I haven't read those ones yet). If that PR is going to come "more later", maybe it would look cleaner to defer the changes to `ncclCommDevIdxMap` to that PR? I don't have a very strong opinion here tho.

Reply:
Yes, totally. I broke the entire change down in 4 chunks in order to make it incremental and "guide" the reviewers through it. If you prefer I can squash them all together